From 9fc7bf708117ef6542948a17afb0fe95fb4d0a01 Mon Sep 17 00:00:00 2001
From: Hans Christian Lonstad <hcl@datarespons.no>
Date: Tue, 28 Apr 2020 09:52:28 +0200
Subject: [PATCH 08/12] Laerdal Sound: Add boards and allow ESAI to handle 32
 bit format

ESAI hw supports 32 bit which is needed for the 1681 TDM operation
---
 sound/soc/fsl/Kconfig    | 28 ++++++++++++++++++++++++++++
 sound/soc/fsl/Makefile   |  5 +++++
 sound/soc/fsl/fsl_esai.c |  3 ++-
 3 files changed, 35 insertions(+), 1 deletion(-)

diff --git a/sound/soc/fsl/Kconfig b/sound/soc/fsl/Kconfig
index b96947d6c026..39d9a1797ceb 100644
--- a/sound/soc/fsl/Kconfig
+++ b/sound/soc/fsl/Kconfig
@@ -596,6 +596,34 @@ config SND_SOC_IMX_XCVR
 	  Say Y if you want to add support for SoC audio on an i.MX board with
 	  an Audio Transceiver (XCVR).
 
+config SND_SOC_LM_IMX_WM8960
+    tristate "SoC Audio support for Laerdal board with wm8960"
+    depends on OF && I2C
+    select SND_SOC_WM8960_LM
+    select SND_SOC_IMX_PCM_DMA
+    select SND_SOC_FSL_SAI
+    select SND_SOC_FSL_UTILS
+    select SND_SOC_IMX_AUDMUX
+    select SND_KCTL_JACK
+    select SND_SOC_FSL_SSI
+    help
+     SoC Audio support for Laerdal boards with WM8960
+     Say Y if you want to add support for SoC audio on an i.MX board with
+     a wm8960 codec.
+
+config SND_SOC_LM_PCM1681
+    tristate "SoC Audio support for Laerdal boards with pcm1681"
+    depends on OF && I2C && LINKBOX_PLUS_PSY
+    select SND_SOC_PCM1681
+    select SND_SOC_IMX_PCM_DMA
+    select SND_SOC_FSL_SAI
+    select SND_SOC_FSL_UTILS
+    select SND_SOC_FSL_ESAI
+    help
+     SoC Audio support for Laerdal boards with pcm1681
+     Say Y if you want to add support for SoC audio on an i.MX board with
+     a 1681 codec.
+
 endif # SND_IMX_SOC
 
 endmenu
diff --git a/sound/soc/fsl/Makefile b/sound/soc/fsl/Makefile
index cec61cc51fcc..7bff5c18344d 100644
--- a/sound/soc/fsl/Makefile
+++ b/sound/soc/fsl/Makefile
@@ -103,6 +103,8 @@ snd-soc-imx-hdmi-objs := imx-hdmi.o
 snd-soc-imx-cdnhdmi-objs := imx-cdnhdmi.o
 snd-soc-imx-rpmsg-objs := imx-rpmsg.o
 snd-soc-imx-xcvr-objs := imx-xcvr.o
+snd-soc-lm-imx-wm8960 := lm-imx-wm8960.o
+snd-soc-lm-pcm1681 := imx-pcm1681.o
 
 obj-$(CONFIG_SND_SOC_EUKREA_TLV320) += snd-soc-eukrea-tlv320.o
 obj-$(CONFIG_SND_SOC_PHYCORE_AC97) += snd-soc-phycore-ac97.o
@@ -132,4 +134,7 @@ obj-$(CONFIG_SND_SOC_IMX_HDMI) += snd-soc-imx-hdmi.o
 obj-$(CONFIG_SND_SOC_IMX_CDNHDMI) += snd-soc-imx-cdnhdmi.o
 obj-$(CONFIG_SND_SOC_IMX_XCVR) += snd-soc-imx-xcvr.o
 
+obj-$(CONFIG_SND_SOC_LM_IMX_WM8960) += snd-soc-lm-imx-wm8960
+obj-$(CONFIG_SND_SOC_LM_PCM1681) += snd-soc-lm-pcm1681
+
 AFLAGS_hdmi_pcm.o := -march=armv7-a -mtune=cortex-a9 -mfpu=neon -mfloat-abi=softfp
diff --git a/sound/soc/fsl/fsl_esai.c b/sound/soc/fsl/fsl_esai.c
index 3426d324acf9..04fb639130c5 100644
--- a/sound/soc/fsl/fsl_esai.c
+++ b/sound/soc/fsl/fsl_esai.c
@@ -20,7 +20,8 @@
 #define FSL_ESAI_FORMATS	(SNDRV_PCM_FMTBIT_S8 | \
 				SNDRV_PCM_FMTBIT_S16_LE | \
 				SNDRV_PCM_FMTBIT_S20_3LE | \
-				SNDRV_PCM_FMTBIT_S24_LE)
+				SNDRV_PCM_FMTBIT_S24_LE) \
+				SNDRV_PCM_FMTBIT_S32_LE)
 
 /**
  * fsl_esai_soc_data: soc specific data
-- 
2.20.1

